src/main/java/jenkins/branch/BranchSource.java

```diff
@@ -34,6 +34,8 @@
 import java.util.List;
 import org.kohsuke.stapler.DataBoundSetter;
 
+import javax.annotation.Nonnull;
```

Comment:
Prefer `edu.umd.cs.findbugs.annotations` over `javax.annotation` as:

1. https://docs.oracle.com/javase/7/docs/api/javax/annotation/package-summary.html does not have `Nonnull`
2. You are not supposed to use the `javax` package space outside of the supervision of a JSR
3. JSR-305 has not made any releases, therefore any artifacts *claiming* to be produced by JSR-305 are just rumours.
4. We do not even know when/if JSR-305 is revived whether they will pick `Nonnull` or `NonNull` or what semantics they will pick for it, so we risk LinkageErrors by using it.
5. The only reason to use `javax.annotation.Nonnull` is if overriding a method interface from Jenkins core that has incorrectly used `javax.annotation.Nonnull`.

As this is an internal usage only, this should - as plugin maintainer - be `edu.umd.cs.findbugs.annotations.NonNull`